File: SWEA/D2/1946. 간단한 압축 풀기/간단한 압축 풀기.py
T = int(input())

for t in range(1, T + 1):
    N = int(input())
    alphabets = []
    for _ in range(N):
        c, n = input().split()
        n = int(n)
        alphabets.append((c, n))

    print(f"#{t}")
    curr_line = 0
    for alphabet, repeats in alphabets:
        for i in range(repeats):
            if curr_line == 10:
                curr_line = 0
                print()
            print(alphabet, end="")
            curr_line += 1
    print()

File: SWEA/D2/1946. 간단한 압축 풀기/간단한 압축 풀기.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int T;
    cin >> T;
    
    for (int t = 1; t <= T; t++) {
        int N;
        cin >> N;
        vector<pair<char, int>> alphabets(N);
        
        for (int i = 0; i < N; i++) {
            cin >> alphabets[i].first >> alphabets[i].second;
        }

        cout << "#" << t << "\n";
        int curr_line = 0;
        for (auto alphabet : alphabets) {
            for (int i = 0; i < alphabet.second; i++) {
                if (curr_line == 10) {
                    cout << "\n";
                    curr_line = 0;
                }
                cout << alphabet.first;
                curr_line++;
            }
        }
        cout << "\n";
    }
    return 0;
}